ranger-discovery-bundle/src/main/java/io/appform/ranger/discovery/bundle/id/generator/IdGeneratorBase.java

```diff
@@ -0,0 +1,160 @@
+package io.appform.ranger.discovery.bundle.id.generator;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import io.appform.ranger.discovery.bundle.id.Domain;
+import io.appform.ranger.discovery.bundle.id.Id;
+import io.appform.ranger.discovery.bundle.id.constraints.IdValidationConstraint;
+import io.appform.ranger.discovery.bundle.id.formatter.IdFormatter;
+import io.appform.ranger.discovery.bundle.id.nonce.NonceGeneratorBase;
+import io.appform.ranger.discovery.bundle.id.request.IdGenerationRequest;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.joda.time.format.DateTimeFormatter;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+/**
+ * Base Id Generator
+ */
+@SuppressWarnings("unused")
+@Slf4j
+public class IdGeneratorBase {
```

Comment:
All methods in this class are final. Looks like this would be use case of composition than inheritance